infra/lib/shared/lambda.ts

```diff
@@ -0,0 +1,193 @@
+      evaluationPeriods: 1,
+      alarmDescription: "Alarm if we get an OOM error from the Lambda function",
+      treatMissingData: cloudwatch.TreatMissingData.NOT_BREACHING,
+    });
+    props.alarmSnsAction && alarm.addAlarmAction(props.alarmSnsAction);
+  }
+
+  return lambda;
+}
+
+export interface RetryLambdaProps extends Omit<LambdaProps, "entry"> {
+  entry?: string;
+  sourceQueue: Queue;
+  destinationQueue: Queue;
+}
+
+/**
+ * Creates a lambda to retry messages from a dead letter queue.
+ *
+ * @param props.entry The source code entry point, defaults to the Reenque's file name.
+ */
+export function createRetryLambda(props: RetryLambdaProps): Lambda {
+  const config = getConfig();
+  const retryLambdaName = props.name + "_Retry_";
+  const retryLambda = createLambda({
+    ...props,
+    name: retryLambdaName,
+    entry: props.entry ?? `${pathToLambdas}/sqs-to-sqs/index.js`,
+    envVars: {
+      ...props.envVars,
+      ENV_TYPE: config.environmentType,
+      ...(config.lambdasSentryDSN ? { SENTRY_DSN: config.lambdasSentryDSN } : {}),
+      SOURCE_QUEUE: props.sourceQueue.queueUrl,
+      DESTINATION_QUEUE: props.destinationQueue.queueUrl,
+    },
+  });
+  props.sourceQueue.grantConsumeMessages(retryLambda);
+  props.destinationQueue.grantSendMessages(retryLambda);
+  return retryLambda;
+}
+
+// TODO validate this works
+// export type ScheduledLambdaProps = Omit<LambdaProps, "entry"> & {
+//   scheduleExpression: string | string[];
+// } & (
+//     | {
+//         entry: string;
+//       }
+//     | {
+//         entry?: never;
+//         url: string;
+//       }
+//   );
+
+// /**
+//  * Creates a lambda that is triggered by a schedule.
+//  *
+//  * @param props.scheduleExpression: "Minutes Hours Day-of-month Month Day-of-week Year", see more
+//  *    here: https://docs.aws.amazon.com/lambda/latest/dg/services-cloudwatchevents-expressions.html
+//  * @param props.url The url to call when the lambda is triggered -
+//  */
+// export function createScheduledLambda(props: ScheduledLambdaProps): Lambda {
+//   const lambdaFn = createLambda({
+//     ...props,
+//     ...(props.entry != null
+//       ? {
+//           entry: props.entry,
+//         }
+//       : {
+//           entry: `${pathToLambdas}/scheduled/index.js`,
+//           envVars: {
+//             ...props.envVars,
+//             ...(props.url ? { URL: props.url } : {}),
+//           },
+//         }),
+//   });
+
+//   if (typeof props.scheduleExpression === "string") {
+//     const eventRule: Rule = buildEventRule(
+//       props.stack,
+//       `${props.name}Rule`,
+//       props.scheduleExpression
+//     );
+//     eventRule.addTarget(new LambdaFunction(lambdaFn));
+//   } else {
+//     props.scheduleExpression.forEach((schedule, i) => {
+//       const eventRule: Rule = buildEventRule(props.stack, `${props.name}Rule_${i}`, schedule);
+//       eventRule.addTarget(new LambdaFunction(lambdaFn));
+//     });
+//   }
+
+//   return lambdaFn;
```

Comment:
leftover?

Reply:
I can remove, but had this code ready and thought it was better to leave as reference for when we need to implement than starting from scratch.

But can defs remove, I see the problem with commented out code.

Reply:
ah okay - I don't have strong feelings about commented out code snippets, just making sure this was intentional